test/sqllogictest/temporal.slt

```diff
@@ -0,0 +1,45 @@
+# Copyright Materialize, Inc. All rights reserved.
+#
```

Comment:
This query sometimes fails with an undescripherable error message:

```
DROP TABLE IF EXISTS temporal;
CREATE TABLE temporal (start_time bigint);
INSERT INTO temporal VALUES (1);
select * from temporal where start_time > mz_logical_timestamp() as of 1;
ERROR:  Timestamp (1) is not valid for all inputs: [(User(41), Some(Antichain { elements: [1612787220000] }))]
```
and sometimes succeeeds:

```
DROP TABLE IF EXISTS temporal;
CREATE TABLE temporal (start_time bigint);
INSERT INTO temporal VALUES (1);
materialize=> select * from temporal where start_time > mz_logical_timestamp() as of 1;
 start_time 
------------
(0 rows)
```

Reply:
I *think* this is a standard and expected error. It is returned when a query requests a timestamp that is no longer valid. Materialize's `TABLE` source compacts itself according to wall clock time, which means that if you can get the query in before compaction happens then it works, and if you want perhaps even just a brief moment, the table will have discarded enough history that `as of 1` will error.

I think you should be able to work around this with `--logical-compaction-window off` which will disable compaction. Otherwise, using tables is probably going to be awkward with testing, as they a have non-deterministic relationship to time.

Reply:
Thank you, understood.